src/herder/Upgrades.cpp

```diff
@@ -1388,14 +1388,34 @@ ConfigUpgradeSetFrame::upgradeNeeded(LedgerSnapshot const& ls) const
 }
 
 void
-ConfigUpgradeSetFrame::applyTo(AbstractLedgerTxn& ltx) const
+ConfigUpgradeSetFrame::applyTo(AbstractLedgerTxn& ltx, Application& app) const
 {
+    bool writeLiveSorobanStateSizeWindow = false;
     for (auto const& updatedEntry : mConfigUpgradeSet.updatedEntry)
     {
         LedgerKey key(LedgerEntryType::CONFIG_SETTING);
         auto const id = updatedEntry.configSettingID();
         key.configSetting().configSettingID = id;
-        ltx.load(key).current().data.configSetting() = updatedEntry;
+        auto& currentEntry = ltx.load(key).current().data.configSetting();
+        if (currentEntry.configSettingID() ==
+                ConfigSettingID::CONFIG_SETTING_STATE_ARCHIVAL &&
+            currentEntry.stateArchivalSettings()
+                    .liveSorobanStateSizeWindowSampleSize !=
+                updatedEntry.stateArchivalSettings()
+                    .liveSorobanStateSizeWindowSampleSize)
+        {
+            writeLiveSorobanStateSizeWindow = true;
+        }
+        currentEntry = updatedEntry;
+    }
+
+    if (writeLiveSorobanStateSizeWindow)
+    {
+        SorobanNetworkConfig networkConfig;
+        networkConfig.loadFromLedger(
+            ltx, app.getConfig().CURRENT_LEDGER_PROTOCOL_VERSION,
```

Comment:
It still seems foot-gunny that we still call `maybeUpdateBucketListWindowSize` via `loadFromLedger`. With this change, I believe this is the only place we will need to call `maybeUpdateBucketListWindowSize`. Can we remove it from `loadFromLedger` and call it explicitely here? This lets us simplify the logic a bit too. We can move the call to `writeliveSorobanStateSizeWindow` into `maybeUpdateBucketListWindowSize` as well.

Reply:
As far as I can tell this change has been made (I don't entirely understand the logic surrounding it, but the edit appears to match what you requested) so I'm approving in order to keep the merge train running. Apologies if this was in error!